src/cider/type/plan/CaseExpr.cpp

```diff
@@ -0,0 +1,278 @@
+  if (expr_pair_list.size() != rhs_ce.get_expr_pair_list().size()) {
+    return false;
+  }
+  if ((else_expr == nullptr && rhs_ce.get_else_expr() != nullptr) ||
+      (else_expr != nullptr && rhs_ce.get_else_expr() == nullptr)) {
+    return false;
+  }
+  auto it = rhs_ce.get_expr_pair_list().cbegin();
+  for (auto p : expr_pair_list) {
+    if (!(*p.first == *it->first) || !(*p.second == *it->second)) {
+      return false;
+    }
+    ++it;
+  }
+  return else_expr == nullptr ||
+         (else_expr != nullptr && *else_expr == *rhs_ce.get_else_expr());
+}
+
+std::string CaseExpr::toString() const {
+  std::string str{"CASE "};
+  for (auto p : expr_pair_list) {
+    str += "(";
+    str += p.first->toString();
+    str += ", ";
+    str += p.second->toString();
+    str += ") ";
+  }
+  if (else_expr) {
+    str += "ELSE ";
+    str += else_expr->toString();
+  }
+  str += " END ";
+  return str;
+}
+
+void CaseExpr::find_expr(bool (*f)(const Expr*),
+                         std::list<const Expr*>& expr_list) const {
+  if (f(this)) {
+    add_unique(expr_list);
+    return;
+  }
+  for (auto p : expr_pair_list) {
+    p.first->find_expr(f, expr_list);
+    p.second->find_expr(f, expr_list);
+  }
+  if (else_expr != nullptr) {
+    else_expr->find_expr(f, expr_list);
+  }
+}
+
+void CaseExpr::collect_rte_idx(std::set<int>& rte_idx_set) const {
+  for (auto p : expr_pair_list) {
+    p.first->collect_rte_idx(rte_idx_set);
+    p.second->collect_rte_idx(rte_idx_set);
+  }
+  if (else_expr != nullptr) {
+    else_expr->collect_rte_idx(rte_idx_set);
+  }
+}
+
+void CaseExpr::collect_column_var(
+    std::set<const ColumnVar*, bool (*)(const ColumnVar*, const ColumnVar*)>& colvar_set,
+    bool include_agg) const {
+  for (auto p : expr_pair_list) {
+    p.first->collect_column_var(colvar_set, include_agg);
+    p.second->collect_column_var(colvar_set, include_agg);
+  }
+  if (else_expr != nullptr) {
+    else_expr->collect_column_var(colvar_set, include_agg);
+  }
+}
+
+void CaseExpr::check_group_by(
+    const std::list<std::shared_ptr<Analyzer::Expr>>& groupby) const {
+  for (auto p : expr_pair_list) {
+    p.first->check_group_by(groupby);
+    p.second->check_group_by(groupby);
+  }
+  if (else_expr != nullptr) {
+    else_expr->check_group_by(groupby);
+  }
+}
+
+JITExprValue& CaseExpr::codegen(CodegenContext& context) {
+  JITFunction& func = *context.getJITFunction();
+  const auto& expr_pair_list = get_expr_pair_list();
+  const auto& else_expr = get_else_ref();
+  CHECK_GT(expr_pair_list.size(), 0);
+  const auto case_ti = get_type_info();
+  if (case_ti.is_integer() || case_ti.is_time() || case_ti.is_decimal() ||
+      case_ti.is_fp() || case_ti.is_boolean()) {
+    const auto type =
+        case_ti.is_decimal() ? decimal_to_int_type(case_ti) : case_ti.get_type();
+    JITValuePointer value = func.createVariable(getJITTag(type), "case_when_value_init");
+    *value = func.createLiteral(getJITTag(type), 0);
```

Comment:
L226 and L227 should can be merged as `JITValuePointer value = func.createVariable(getJITTag(type), "case_when_value_init", 0);`?

Reply:
Done.

Reply:
Seems these lines can't be changed. Try reverting back and fix code style/lint issue at the same time.

Reply:
Done.